### PR TITLE
Handling session loss for non Session-Lock-Lost exceptions + Increasing AMQP nuget reference to 2.1.0

### PR DESCRIPTION
**Pull request by nemakam**
## Description

Throwing SessionLockLost exception when session is lost due to errors other than SessionLockLostException thrown from the server (eg: communication exceptions).

Also, increased Microsoft.Azure.Amqp nuget reference to 2.1.0

### CODE REVIEW

File: src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs [hunk 1]
```diff
@@ -791,9 +791,9 @@ void OnSessionReceiverLinkClosed(object sender, EventArgs e)
                     a => receiveLink.EndReceiveMessages(a, out amqpMessages),
                     this).ConfigureAwait(false);
 
-                    if (receiveLink.TerminalException != null)
+                    if (receiveLink.GetInnerException() != null)
                     {
-                        throw receiveLink.TerminalException;
+                        throw receiveLink.GetInnerException();
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs]: >GetInnerException [](start = 42, length = 17)

why twice? can we just keep the exception check and throw that exception

Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs]: Changed the logic. Please take a look

File: src/Microsoft.Azure.ServiceBus/MessagingEventSource.cs [hunk 2]
```diff
@@ -1210,12 +1211,11 @@ void GetRulesException(string clientId, string exception)
...
 
         [NonEvent]
-        public void CreatingNewLink(string clientId, bool isSessionReceiver, string sessionId, bool isRequestResponseLink, Error linkError)
+        public void CreatingNewLink(string clientId, bool isSessionReceiver, string sessionId, bool isRequestResponseLink, Exception linkException)
         {
             if (this.IsEnabled())
             {
-                string linkErrorString = linkError != null ? linkError.Condition.Value : string.Empty;
-                this.CreatingNewLink(clientId, isSessionReceiver, sessionId ?? string.Empty, isRequestResponseLink, linkErrorString);
+                this.CreatingNewLink(clientId, isSessionReceiver, sessionId ?? string.Empty, isRequestResponseLink, linkException.ToString());
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/MessagingEventSource.cs]: >linkException. [](start = 116, length = 14)

null check?

File: src/Microsoft.Azure.ServiceBus/MessagingEventSource.cs [hunk 3]
```diff
@@ -1226,19 +1226,18 @@ void CreatingNewLink(string clientId, bool isSessionReceiver, string sessionId,
...
 
         [NonEvent]
-        public void SessionReceiverLinkClosed(string clientId, string sessionId, Error linkError)
+        public void SessionReceiverLinkClosed(string clientId, string sessionId, Exception linkException)
         {
             if (this.IsEnabled())
             {
-                string linkErrorString = linkError != null ? linkError.Condition.Value : string.Empty;
-                this.SessionReceiverLinkClosed(clientId, sessionId ?? string.Empty, linkErrorString);
+                this.SessionReceiverLinkClosed(clientId, sessionId ?? string.Empty, linkException.ToString());
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/MessagingEventSource.cs]: >linkException [](start = 84, length = 13)

null check?

**Pull request merged**